torch/_dynamo/compiled_autograd.py

```diff
@@ -183,3 +196,108 @@ def disable():
+    import torch._functorch.aot_autograd
+
+    assert len(args) == 1
+    grad = args[0]
+    assert isinstance(grad, torch.Tensor)
+
+    mode = _get_current_dispatch_mode()
+    if isinstance(fn, functools.partial):
+        fn.__name__ = fn.func.__name__  # type: ignore[attr-defined]
+    original_fn = fn
+    if reenter:
+        # If the reenter flag is set, we wrap the original fn in dynamo_interceding_fn_wrapper
+        # and write that to the graph. This produces an aot_autograd graph during backwards that
+        # points to dynamo_interceding_fn_wrapper. Then, during compiled autograd, we use
+        # dynamo_interceding_fn_wrapper to _invoke_in_backward the original fn under dynamo. The actual
+        # dynamo part of dynamo_interceding_fn_wrapper happens during compiled autograd.
+        fn = functools.partial(dynamo_interceding_fn_wrapper, fn=fn)
+        fn.__name__ = fn.func.__name__
+
+    is_func = torch._is_functional_tensor(grad)
+    if is_func:
+        grad = torch._functorch.aot_autograd.from_fun(grad)
+
+    proxy_args = pytree.tree_map(mode.tracer.unwrap_proxy, (grad,))
+    out_proxy = mode.tracer.create_proxy(
+        "call_function", fn, proxy_args, {}, name="invocation"
+    )
+    grad = track_tensor_tree(grad, out_proxy, constant=None, tracer=mode.tracer)
+
+    # We have a little shortcut here, wherein we DO NOT yet run a meta func, and so
+    # we take on an assumption that input and output meta matches. As such, we must introduce
+    # a runtime assert
+    proxy_args = pytree.tree_map(
+        mode.tracer.unwrap_proxy, (grad, grad.size(), grad.stride(), grad.dtype)
+    )
+    out_proxy = mode.tracer.create_proxy(
+        "call_function",
+        torch._functional_assert_tensor_metadata,
+        proxy_args,
+        {},
+        name="assert",
+    )
+    grad = track_tensor_tree(grad, out_proxy, constant=None, tracer=mode.tracer)
+    return (
+        pytree.tree_map(torch._functorch.aot_autograd.to_fun, grad) if is_func else grad
+    )
+
+
+@_invoke_in_backward_op.py_impl(FakeTensorMode)
+def inner_fake(*args, fn, reenter):
+    raise RuntimeError("This op should never be invoked here")
+
+
+@_invoke_in_backward_op.py_impl(DispatchKey.CompositeExplicitAutograd)
+def _invoke_in_backward_op_dense(*args, fn, reenter):
+    if reenter:
+        return _invoke_in_backward_op(*args, fn=fn, reenter=False)
+    mode = _get_current_dispatch_mode()
+    assert mode is None, "Mode should never be enabled for CPU/CUDA key"
+    return fn(*args)
+
+
+_invoke_in_backward_op.py_impl(DispatchKey.Autograd)(
+    autograd_not_implemented(_invoke_in_backward_op, deferred_error=True)
+)
+
+
+@_invoke_in_backward_op.py_impl(DispatchKey.Functionalize)
+def _invoke_in_backward_functionalized(*args, fn, reenter):
+    mode = _get_current_dispatch_mode()
+    with _ExcludeDispatchKeyGuard(DispatchKeySet(DispatchKey.Functionalize)):
+        return _invoke_in_backward_op(*args, fn=fn, reenter=reenter)
```

Comment:
does this just mean we ignore functionalization?

Reply:
No, this re-enter the dispatcher and falls through to inner_trace